gps_driver:
  port: "/dev/gps"
  baud: 115200
  useRMC: false # get covariance instead of velocity, see wiki for more info

imu_driver:
  port: "/dev/imu"
  baud: 115200
  frame_id: "imu_link"

brushed_motors:
  controllers:
    - name: "ARM_A"
      open_loop_topic: "open_loop/ra/joint_a"
      joint_data_topic: "joint_data/ra/joint_a"
      driver_voltage: 36.0
      motor_max_voltage: 24.0
      voltage_multiplier: 0.44
      nucleo: 0
      channel: 0
      quad_cpr: -28800.0
      kP: -0.005
      kI: -0.0
      inversion: -1.0
    - name: "ARM_B"
      open_loop_topic: "open_loop/ra/joint_b"
      joint_data_topic: "joint_data/ra/joint_b"
      driver_voltage: 36.0
      motor_max_voltage: 12.0
      nucleo: 0
      channel: 1
      quad_cpr: 4096.0
      kP: -0.1
      kI: -0.0
      inversion: 1.0
    - name: "ARM_C"
      open_loop_topic: "open_loop/ra/joint_c"
      joint_data_topic: "joint_data/ra/joint_c"
      driver_voltage: 36.0
      motor_max_voltage: 24.0
      nucleo: 0
      channel: 1
      quad_cpr: 136800.0
      kP: -0.005
      kI: -0.0
      inversion: 1.0
    - name: "ARM_D"
      open_loop_topic: "open_loop/ra/joint_d"
      joint_data_topic: "joint_data/ra/joint_d"
      driver_voltage: 36.0
      motor_max_voltage: 24.0
      nucleo: 1
      channel: 1
      quad_cpr: 72000.0
      kP: -0.005
      kI: -0.0
      inversion: 1.0
    - name: "ARM_E"
      open_loop_topic: "open_loop/ra/joint_e"
      joint_data_topic: "joint_data/ra/joint_e"
      driver_voltage: 36.0
      motor_max_voltage: 24.0
      nucleo: 2
      channel: 0
      quad_cpr: -72000.0
      kP: -0.005
      kI: -0.0
      inversion: 1.0
    - name: "ARM_F"
      open_loop_topic: "open_loop/ra/joint_f"
      joint_data_topic: "joint_data/ra/joint_f"
      driver_voltage: 36.0
      motor_max_voltage: 24.0
      nucleo: 2
      channel: 1
      quad_cpr: -2048.0
      kP: 0.05
      kI: 0.0
      inversion: 1.0
    - name: "HAND_FINGER"
      open_loop_topic: "open_loop/hand/finger"
      driver_voltage: 12.0
      motor_max_voltage: 12.0
      nucleo: 2
      channel: 2
      inversion: 1.0
    - name: "HAND_GRIP"
      open_loop_topic: "open_loop/hand/grip"
      driver_voltage: 12.0
      motor_max_voltage: 12.0
      nucleo: 1
      channel: 2
      inversion: 1.0
  i2c_device_file: "/dev/i2c-1"

odrive:
  axis:
    left: 0
    right: 1
  config:
    current_lim: 4
    watchdog_timeout: 0.1
  ids:
    front: "335D36623539"
    middle: "205535833056"
    back: "2066377F5753"
  ratio:
    meters_to_turns_ratio_left: -39.793
    meters_to_turns_ratio_right: 39.793
    turns_to_raw_ratio_left: -50.0
    turns_to_raw_ratio_right: 50.0
  ros:
    publish_rate_hz: 20
